HW0/P5/P5.txt

```diff
@@ -1 +1,60 @@
-# Your answers here.
+1. It would take 255 seconds since that is the number of consecutive pairs.
+
+2. Yes. There are a total of 8 people counting, so each person takes on 256 / 8 = 32 bags.
+   It takes each person 31 seconds to verify his/her bags. Then they are left with 8 numbers to add.
+   Now, they can have 4 people each adding 2 numbers, this takes 1 more second. After which there 
+   are 4 numbers left. Again they will have 2 people each counting 2 numbers which takes another second.
+   Finally, there are only 2 numbers left and the cashier can verify this, which takes 1 last second. 
+   Thus in total they can do this in: 31 + 1 + 1 + 1 = 34 seconds.
+   
+3. Even if there are infinite number of employees, only 128 of them can actually do any work since
+   there are only 128 pairs of numbers to add initially. This first step takes 1 second. In the next
+   step, there is half the number of pairs left (or 64), so following this recursion, it would take a total of
+   log(256) * 1 = 8 seconds to verify the sum.
+   
+4. Let k be such that 2^k >= N >= 2^(k-1). Then with similar reasoning to part (3), it will take 
+   the store k seconds to verify the sum. If N is not a power of 2, we can make the process exactly the same as
+   in part (3) by adding "virtual" bags with zero counts so that the total number of bags is 2^k. Then the same 
+   counting procedure follows. Note that k = ceil(log(N)).
+   
+5. In this case, they can start off with 256 people each having a bag (assuming infinite number of employees). 
+   Then in the first round, half of them need to communicate the bag numbers with the other half so the latter 
+   each has 2 numbers to add. This round takes 2 seconds to complete. Similarly for every consequent round. 
+   Since there are 8 rounds in total (in round 8, 2 people each has 1 bag, so one needs to communicate to the 
+   other, and the other can add the final two numbers), it will take 2 * 8 = 16 seconds. 
+   
+6. If each cashier takes 1 bag then the total time to hand all 256 bags to the 256 cashiers would be 256 seconds.
```

Comment:
I think the cashiers that get their bags first can start summing while you continue to hand more bags out. As a result, by the time you hand the last bag out the first 255 people can be mostly done with their summing.